packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,57 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/foundation.dart' as foundation;
+/// import 'package:flutter/gestures.dart' as gestures;
+/// import 'package:flutter/rendering.dart' as rendering;
+/// import 'package:flutter/scheduler.dart' as scheduler;
+/// import 'package:flutter/services.dart' as services;
+/// import 'package:flutter/widgets.dart' as widgets;
+///
+/// class MyImageCache extends widgets.ImageCache {
+///   @override
+///   void clear() {
+///     print("Clearing cache!");
```

Comment:
nit: Maybe instead of having an example method with a strange print just have a comment in this class stating that this is were you implement your custom cache?

Reply:
I placed an example override at the request of Dan.  His point was that it makes MyImageCache more noticeable and the overriding of behavior explicit.

Reply:
I see. It may make sense to still have a comment in this class telling people that this is the place where they can do their custom image caching.